exercises/affine-cipher/tests/affine-cipher.rs

```diff
@@ -0,0 +1,152 @@
+use affine_cipher::*;
+
+#[test]
+fn encode_yes() {
+    assert_eq!(encode("yes", 5, 7).unwrap(), "xbt")
+}
+
+#[test]
+#[ignore]
+fn encode_no() {
+    assert_eq!(encode("no", 15, 18).unwrap(), "fu")
+}
+
+#[test]
+#[ignore]
+fn encode_omg() {
+    assert_eq!(encode("OMG", 21, 3).unwrap(), "lvz")
+}
+
+#[test]
+#[ignore]
+fn encode_o_m_g() {
+    assert_eq!(encode("O M G", 25, 47).unwrap(), "hjp")
+}
+
+#[test]
+#[ignore]
+fn encode_mindblowingly() {
+    assert_eq!(
+        encode("mindblowingly", 11, 15).unwrap(),
+        "rzcwa gnxzc dgt"
+    )
+}
+
+#[test]
+#[ignore]
+fn encode_numbers() {
+    assert_eq!(
+        encode("Testing,1 2 3, testing.", 3, 4).unwrap(),
+        "jqgjc rw123 jqgjc rw"
+    )
+}
+
+#[test]
+#[ignore]
+fn encode_deep_thought() {
+    assert_eq!(
+        encode("Truth is fiction", 5, 17).unwrap(),
+        "iynia fdqfb ifje"
+    )
+}
+
+#[test]
+#[ignore]
+fn encode_all_the_letters() {
+    assert_eq!(
+        encode("The quick brown fox jumps over the lazy dog.", 17, 33).unwrap(),
+        "swxtj npvyk lruol iejdc blaxk swxmh qzglf"
+    )
+}
+
+#[test]
+#[ignore]
+fn encode_with_a_not_coprime_to_m() {
+    let expected_error = AffineCipherError::NotCoprime(6);
+    match encode("This is a test.", 6, 17) {
+        Err(AffineCipherError::NotCoprime(6)) => (),
```

Comment:
```suggestion
        Err(expected_error) => (),
```

Note that this will require `#[derive(Clone,Copy)]` on `AffineCipherError`.

Reply:
Will do.

Reply:
The suggested approach doesn't quite work -- it binds a local `expected_error` variable to whatever `Err()` contains rather than matching against the outer `expected_error` -- but it did get me thinking about this a little differently.

While we can't pattern match on variables, we *can* pattern match on constants (provided they implement `Eq` and `PartialEq`), so I changed `expected_error` into a `const` and added the appropriate derives.

Hopefully that's a little bit closer to what you're looking for.

Reply:
Yes, it is; that looks great! Sorry for having lead you astray with the variable binding.